python/nav/portadmin/snmp/cisco.py

```diff
@@ -300,5 +315,88 @@ def get_dot1x_enabled_interfaces(self):
             for oid, state in self._bulkwalk(self.dot1xPortAuth)
         }
 
+    def get_poe_state_options(self) -> Sequence[PoeState]:
+        """Returns the available options for enabling/disabling PoE on this netbox"""
+        options_list = [
+            self.POE_AUTO,
+            self.POE_STATIC,
+            self.POE_LIMIT,
+            self.POE_DISABLE,
+        ]
+        return options_list
+
+    @translate_protocol_errors
+    def set_poe_state(self, interface: manage.Interface, state: PoeState):
+        """Set state for enabling/disabling PoE on this interface.
+        Available options should be retrieved using `get_poe_state_options`
+        """
+        unit_number, interface_number = self._get_poe_indexes_for_interface(interface)
+        oid_with_unit_number = self.POEENABLE + f".{unit_number}"
+        try:
+            self._set_netbox_value(
+                oid_with_unit_number, interface_number, 'i', state.state
+            )
+        except SnmpError as error:
+            _logger.error('Error setting poe state: %s', error)
+            raise
+        except ValueError as error:
+            _logger.error('%s is not a valid option for poe state', state)
+            raise
+
+    def _get_poe_indexes_for_interface(
+        self, interface: manage.Interface
+    ) -> Tuple[int, int]:
+        """Returns the unit number and interface number for the given interface"""
+        try:
+            poeport = manage.POEPort.objects.get(interface=interface)
+        except manage.POEPort.DoesNotExist:
+            raise POEIndexNotFoundError(
+                "This interface does not have PoE indexes defined"
+            )
+        unit_number = poeport.poegroup.index
+        interface_number = poeport.index
+        return unit_number, interface_number
+
+    def get_poe_states(
+        self, interfaces: Sequence[manage.Interface] = None
+    ) -> Dict[int, Optional[PoeState]]:
+        """Retrieves current PoE state for interfaces on this device.
+
+        :param interfaces: Optional sequence of interfaces to filter for, as fetching
```

Comment:
If `interfaces` is in fact optional, shouldn't it also be type annotated as such?  (maybe the existing code doesn't, and maybe it should)

Reply:
This is how its done in other parts of the handler.py class, so I did the same for consistency. But it could very much make sense to add `Optional` typing

Reply:
We can go back and fix the original type annotations in a different PR as well.  I think the ManagementHandler code was one of my earliest experiences with adding type annotations to everything, so I still had a lot to learn :)